### ISSUE TITLE
[Bug]: Crash app at https://www.yodobashi.com/

### ISSUE THREAD

**Issue by lydhvmo** [MetaMask/metamask-mobile#9018]
### Describe the bug

Clicking on the search section on https://www.yodobashi.com/ caused the app to crash on Android

### Expected behavior

no crashes

### Screenshots/Recordings

![image](https://github.com/MetaMask/metamask-mobile/assets/147371557/535831dd-966b-4448-9ef4-bb10ae6de3db)

### Steps to reproduce

Go to ['yodobashi'](https://www.yodobashi.com/)
Click on search at home screen

### Version

7.17.1

### Build type

None

### Device

All smart phone samsung

### Operating system

Android

### Additional context

After a bug occurred, the app could no longer be used even if I cleared the app's cache
Root cause: https://github.com/square/okhttp/issues/6897
decodeHeaderAsJavaNetCookies(JavaNetCookieJar.kt:98)

**Comment by nghia-tran-alobridge** [comment]
Detail:
OKHTTP_VERSION (react-native v0.71.15) : https://github.com/facebook/react-native/blob/5761ad6437b29dc559fce0914f2a8e71d02d25b3/ReactAndroid/gradle.properties#L17

**Comment by gauthierpetetin** [comment]
Hi @lydhvmo , thanks for reporting this issue. How did you manage to identify https://github.com/square/okhttp/issues/6897 as the root cause?

**Comment by nghia-tran-alobridge** [comment]
Hi @gauthierpetetin.
We,@lydhvmo and I, built the Metamask app and found this message in the Android Studio - Logcat

<img width="1260" alt="Screenshot 2024-03-25 at 13 55 40" src="https://github.com/MetaMask/metamask-mobile/assets/129494061/29248c51-25ac-45fb-b1a9-b676e38e0f52">


```
E  FATAL EXCEPTION: OkHttp Dispatcher
	Process: io.metamask.flask, PID: 23779
	java.lang.StringIndexOutOfBoundsException: String index out of range: -1
		at java.lang.String.substring(String.java:1975)
		at okhttp3.JavaNetCookieJar.decodeHeaderAsJavaNetCookies(JavaNetCookieJar.kt:98)
		at okhttp3.JavaNetCookieJar.loadForRequest(JavaNetCookieJar.kt:59)
		at com.facebook.react.modules.network.ReactCookieJarContainer.loadForRequest(ReactCookieJarContainer.java:44)
		at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:74)
		at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
		at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76)
		at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
		at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201)
		at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:517)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
		at java.lang.Thread.run(Thread.java:764)
```

https://github.com/square/okhttp/blob/parent-4.9.2/okhttp-urlconnection/src/main/kotlin/okhttp3/JavaNetCookieJar.kt#L98

**Comment by jccazeaux** [comment]
Hi,

we have the exact same problem on our app. After visiting a website in webview the app crashed until we clear all data or wait for 1 day. Is there any fix or workaround?

**Comment by NicolasMassart** [comment]
@jccazeaux could you please share the link to your website if it's publicly available?

**Comment by joaoloureirop** [comment]
```
08-29 16:48:51.652 26816 26945 E AndroidRuntime: FATAL EXCEPTION: hj.a0 Dispatcher
08-29 16:48:51.652 26816 26945 E AndroidRuntime: Process: io.metamask, PID: 26816
08-29 16:48:51.652 26816 26945 E AndroidRuntime: java.lang.StringIndexOutOfBoundsException: begin 1, end 0, length 1
08-29 16:48:51.652 26816 26945 E AndroidRuntime:        at java.lang.String.checkBoundsBeginEnd(String.java:4466)
08-29 16:48:51.652 26816 26945 E AndroidRuntime:        at java.lang.String.substring(String.java:2517)
08-29 16:48:51.652 26816 26945 E AndroidRuntime:        at hj.x.e(JavaNetCookieJar.kt:74)
08-29 16:48:51.652 26816 26945 E AndroidRuntime:        at hj.x.c(JavaNetCookieJar.kt:115)
08-29 16:48:51.652 26816 26945 E AndroidRuntime:        at com.facebook.react.modules.network.k.c(ReactCookieJarContainer.java:5)
08-29 16:48:51.652 26816 26945 E AndroidRuntime:        at nj.a.a(BridgeInterceptor.kt:131)
08-29 16:48:51.652 26816 26945 E AndroidRuntime:        at nj.g.a(RealInterceptorChain.kt:167)
08-29 16:48:51.652 26816 26945 E AndroidRuntime:        at nj.j.a(RetryAndFollowUpInterceptor.kt:35)
08-29 16:48:51.652 26816 26945 E AndroidRuntime:        at nj.g.a(RealInterceptorChain.kt:167)
08-29 16:48:51.652 26816 26945 E AndroidRuntime:        at mj.e.t(RealCall.kt:114)
08-29 16:48:51.652 26816 26945 E AndroidRuntime:        at mj.e$a.run(RealCall.kt:52)
08-29 16:48:51.652 26816 26945 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
08-29 16:48:51.652 26816 26945 E AndroidRuntime:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
08-29 16:48:51.652 26816 26945 E AndroidRuntime:        at java.lang.Thread.run(Thread.java:1012)
```
Crash happening on MetaMask v7.29.0(1397) from Play Store

Device: Xiaomi 11 Lite 5G NE on Xiaomi HyperOS ver 1.0.5.0 (Android 14)

Steps taken to reproduce:
- Open inApp Browser
- Navigate to `https://www.yodobashi.com`
- Press browser tabs icon at the bottom

https://github.com/user-attachments/assets/3d33cc34-24ef-4dc2-a098-36d3623c800e

**Comment by jccazeaux** [comment]
The crash is caused by this issue in okhttp https://github.com/square/okhttp/issues/6721

It was detected in 06/2021 on 4.9. It was backported in 3.12 but oddly non included in 4.9. After some discussions https://github.com/square/okhttp/issues/8527) with authors they won't do it and ask to go for 5.0.0-aplha wich includes the patch.

**Comment by joaoloureirop** [comment]
Unable to reproduce on android emulator, API 34

**Comment by joaoloureirop** [comment]
Worth mentioning that this website sets third-party cookies which will be blocked in the future

<img width="652" alt="image" src="https://github.com/user-attachments/assets/96d44a1a-614c-4eec-bba4-9147db279c22">

**Comment by joaoloureirop** [comment]
Blocked: fix only after okhttp update

**Comment by sethkfman** [comment]
Issue tested and addressed on 7.61.0
